wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/ParallelCommandGroup.java

```diff
@@ -51,9 +52,20 @@ public final void addCommands(Command... commands) {
     CommandScheduler.getInstance().registerComposedCommands(commands);
 
     for (Command command : commands) {
-      if (!Collections.disjoint(command.getRequirements(), getRequirements())) {
+      var currentRequirements = getRequirements();
+      if (!Collections.disjoint(command.getRequirements(), currentRequirements)) {
+        String requirementsStr = command.getRequirements()
+          .stream()
```

Comment:
The Stream API is really inefficient (for these use cases) and is generally avoided in the WPILib codebase

Reply:
I feel like it’s probably fine to use the stream api here, even if it’s slow, because this code will only run when an error happens anyways

Reply:
Keeping the Java and C++ implementations similar will make it easier to maintain, though.